clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -539,20 +545,28 @@ public boolean isUpdateDue(long now) {
         public long maybeUpdate(long now) {
             // should we update our metadata?
             long timeToNextMetadataUpdate = metadata.timeToNextUpdate(now);
-            long timeToNextReconnectAttempt = Math.max(this.lastNoNodeAvailableMs + metadata.refreshBackoff() - now, 0);
             long waitForMetadataFetch = this.metadataFetchInProgress ? Integer.MAX_VALUE : 0;
```

Comment:
It seems a little inconsistent that we return `Integer.MAX_VALUE` here when a metadata fetch is in progress, yet we return `requestTimeoutMs` after sending the actual metadata fetch. Maybe this should be `requestTimeoutMs` also?